homeassistant/components/huawei_lte/quality_scale.yaml

```diff
@@ -0,0 +1,76 @@
+rules:
+  # Bronze
+  action-setup: done
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: done
+  entity-event-setup: done
+  entity-unique-id: done
+  has-entity-name: done
+  runtime-data: todo
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
+
+  # Silver
+  action-exceptions: todo
+  config-entry-unloading: done
+  docs-configuration-parameters: done
+  docs-installation-parameters: done
+  entity-unavailable: done
+  integration-owner: done
+  log-when-unavailable:
+    status: todo
+    comment: Kind of done, but to be reviewed, there's probably room for improvement. Maybe address this when converting to use a data update coordinator. See also https://github.com/home-assistant/core/issues/55495
+  parallel-updates: todo
+  reauthentication-flow: done
+  test-coverage: todo
+
+  # Gold
+  devices: done
+  diagnostics:
+    status: todo
+    comment: https://github.com/home-assistant/core/pull/131085
+  discovery-update-info: done
+  discovery: done
+  docs-data-update: todo
+  docs-examples: todo
+  docs-known-limitations: todo
+  docs-supported-devices: done
+  docs-supported-functions:
+    status: todo
+    comment: Some info exists, but there's room for improvement.
+  docs-troubleshooting: todo
+  docs-use-cases: todo
+  dynamic-devices: done
+  entity-category: done
+  entity-device-class: done
```

Comment:
```py
class HuaweiLteMobileConnectionBinarySensor(HuaweiLteBaseBinarySensor):
    """Huawei LTE mobile connection binary sensor."""

    _attr_translation_key = "mobile_connection"
    _attr_entity_registry_enabled_default = True

    key = KEY_MONITORING_STATUS
    item = "ConnectionStatus"

    @property
    def is_on(self) -> bool:
        """Return whether the binary sensor is on."""
        return bool(
            self._raw_state
            and int(self._raw_state)
            in (ConnectionStatusEnum.CONNECTED, ConnectionStatusEnum.DISCONNECTING)
        )
```
What is this binary sensor saying? If the device is connected or disconnecting? Why don't we make this just a connected one with a device class for connecitivity?

Reply:
It says if the mobile connection is connected or not. If the raw state of the connection is CONNECTED or DISCONNECTING, the connection is (still) connected, otherwise it's not.

I'm afraid I don't understand the suggestion, would probably need to read up something first.

Reply:
that makes sense, I guess we can do with just the connectivity device class here

Reply:
Binary sensor device class things noted in 8857b75fee73354ae5bb868787a648a0fd96cf42

Reply:
https://github.com/home-assistant/core/pull/143764, status update in 811457ff3b85a55b03116692aca3071c62f1ea26